test/common/alerts_firing.go

```diff
@@ -0,0 +1,135 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+const deadMansSwitch = "DeadMansSwitch"
+
+// alertsTestMetadata contains metadata about the alert
+type alertTestMetadata struct {
+	alertName string
+	podName   string
+	namespace string
+}
+
+// alertsFiringError is a custom alerts error
+type alertsFiringError struct {
+	alertsPending        []alertTestMetadata
+	alertsFiring         []alertTestMetadata
+	deadMansSwitchFiring bool
+}
+
+// Error implements the error interface and returns a readable output message
+func (e *alertsFiringError) Error() string {
+	var str strings.Builder
+	if e.deadMansSwitchFiring {
+		str.WriteString("\nThe following alerts were not fired, but were expected to be firing:")
+		str.WriteString(fmt.Sprintf("\n\talert: %s", deadMansSwitch))
+	}
+	if len(e.alertsPending) != 0 {
+		str.WriteString("\nThe following alerts were pending:")
+		for _, alert := range e.alertsPending {
+			str.WriteString(fmt.Sprintf("\n\talert: %s, for pod: %s, in namespace: %s", alert.alertName, alert.podName, alert.namespace))
+		}
+	}
+	if len(e.alertsFiring) != 0 {
+		str.WriteString("\nThe following alerts were fired:")
+		for _, alert := range e.alertsFiring {
+			str.WriteString(fmt.Sprintf("\n\talert: %s, for pod: %s, in namespace: %s", alert.alertName, alert.podName, alert.namespace))
+		}
+	}
+	return str.String()
+}
+
+// isNotEmpty checks whether or not the error contains firing or pending alerts
+func (e *alertsFiringError) isNotEmpty() bool {
+	return !e.deadMansSwitchFiring || len(e.alertsPending) != 0 || len(e.alertsPending) != 0
+}
+
+func TestIntegreatlyAlertsFiring(t *testing.T, ctx *TestingContext) {
+	var lastError error
+
+	// retry the tests every minute for up to 15 minutes
+	monitoringTimeout := 15 * time.Minute
+	monitoringRetryInterval := 1 * time.Minute
+	err := wait.Poll(monitoringRetryInterval, monitoringTimeout, func() (done bool, err error) {
+		if newErr := getFiringOrPendingAlerts(ctx); newErr != nil {
+			lastError = newErr
+			if _, ok := newErr.(*alertsFiringError); ok {
+				return false, nil
+			}
+			return false, err
+		}
+		return true, nil
```

Comment:
Wouldnt this return "nil" , even though the original variable "err" had a valid error.?

Reply:
Well this would return `nil` from the anonymous function only if `newErr` is nil. If an error is returned from the anonymous function, that error is handled on line 74: https://github.com/integr8ly/integreatly-operator/pull/536/commits/c75dd63873c2bb9c7b3ce6c49e4b9dcf28bbe96e#diff-47e083a29f591bcdd963b0ae626a1012R74

Reply:
Sorry, didnt notice the anonymous method there. My bad.

Reply:
No worries!